backend/de.metas.business/src/main/java/de/metas/bpartner/composite/repository/BPartnerCompositeSaver.java

```diff
@@ -394,6 +401,19 @@ private void saveBPartnerLocation(@NonNull final BPartnerLocationSaveRequest req
 			bpartnerLocationRecord.setSAP_PaymentMethod(partnerLocation.getSapPaymentMethod());
 			bpartnerLocationRecord.setSAP_BPartnerCode(partnerLocation.getSapBPartnerCode());
 
+			// needs to be last, and we generate here to avoid update logs in de.metas.audit.apirequest.ApiAuditLoggable#addTableRecordReferenceLog, if same name is generated
+			final String name = bpartnerLocationRecord.getName();
+			if(name == null || BPARTNER_LOCATION_NAME_DEFAULT.equals(name))
+			{
+				bpartnerLocationRecord.setName(MakeUniqueLocationNameCommand.builder()
+										   .name(bpartnerLocationRecord.getName())
+										   .address(bpartnerLocationRecord.getC_Location())
+										   .companyName(bpartnerBL.getBPartnerName(request.getBpartnerId()))
+										   .existingNames(MakeUniqueLocationNameCommand.getOtherLocationNames(bpartnerLocationRecord.getC_BPartner_ID(), bpartnerLocationRecord.getC_BPartner_Location_ID()))
```

Comment:
this line looks awkard

Reply:
i.e. calling MakeUniqueLocationNameCommand with existingNames having a value computed by MakeUniqueLocationNameCommand itself

Reply:
It's a copy of existing one here `de.metas.bpartner.interceptor.C_BPartner_Location#updateBPLocationName` and used to not get this bplocation itself, if updated